### PR TITLE
Configurable PID file

### PR DESCRIPTION
**Pull request by Exhar**
Allows a user to configure the PID file location without setting ENVs et cetera. Useful for init scripts, for example.

Furthermore, it is best practice to for PID files to be stored in /var/run; and setting the home dir env to /var/run doesn't seem the best solution to comply

### CODE REVIEW

File: lib/oxidized/config.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ def self.load(cmd_opts={})
         asetus.default.rest          = '127.0.0.1:8888' # or false to disable
       asetus.default.vars          = {}             # could be 'enable'=>'enablePW'
       asetus.default.groups        = {}             # group level configuration
+      asetus.default.pid           = '/var/run/oxidized.pid' # location of PID file
```
Comment (ElvinEfendi) [inline lib/oxidized/config.rb]: I think default path should be `File.join(Oxidized::Config::Root, 'pid')`. This lets us run `oxidized` with as minimum permissions as possible. It should not be necessary for the user that runs `oxidized` to have write permission to `/var/run/oxidized.pid`

**Comment by ElvinEfendi** [comment]
> Furthermore, it is best practice to for PID files to be stored in /var/run; and setting the home dir env to /var/run doesn't seem the best solution to comply

Could you explain why it's better to store the `pid` file in `/var/run`? Any article or stackoverflow link would also work.

I like the idea of having everything in `oxidized`'s root directory because this makes things simpler particularly in terms of user permission handling.

**Comment by Exhar** [comment]
http://stackoverflow.com/questions/5173636/must-my-pidfile-be-located-in-var-run

Although the PID could be stored in the homedir (which is as easy as changing the default configured location to '~/.config/oxidized/'), I can image situations where multiple users will try to start Oxidized whilst an instance is running. If this user cannot access a 'public' dir to check if the PID file exists, Oxidized will crash instead of the (already build in) message that Oxidized is running.

It'll crash because the REST port is already in use; this will be the case when users are using the same config, for example because /etc/oxidized/config exists

**Comment by ElvinEfendi** [comment]
I agree that for the daemons that is constrained to run only one instance at the given time, `/var/run` is a better place. But `oxidized` is not a system level application. It should be possible to run multiple instances of it in the same machine. As to it being crashed because of the port is already in use, this is a bug imho. Instead of changing `pid` path, we should let users to configure the `port`.

**Comment by danilopopeye** [comment]
I think putting the `pid` file in `/var/run` makes sense when oxidized is running on some os daemon like init, runit, upstart etc. So I guess the default value be in oxidized root is ok. :+1: 

On account of the port problem, makes more sense to let the user configure it. Can you provide another PR for this? Or let me know and I'll hack one next week if I manage to get some time at work. :smile:

**Comment by roedie** [comment]
Having the pidfile and port configurable isn't a bad thing. Oxidized even look in /etc/oxidized/config for a configuration during start. I can see no other reason than running it as a system daemon, and then it makes sense to have the pidfile in /var/run

The default behavior shouldn't be changed, but it should definitely be configurable.

**Comment by jgroom33** [comment]
Would this change resolve the following issue:
When running as a docker container, I map the config volume to the host:
docker run --name "vrp-oxidized" -d -v /etc/oxidized/vrp:/root/.config/oxidized -p 8101:8888/tcp -t vfnz/oxidized:latest

The pid file is created in that directory, so is stored on the host.  When the docker container is stopped it cannot restart until the pid file is deleted on the host.

**Comment by ytti** [comment]
@Exhar can you change it so that by default it's same dir as currently?

**Comment by Exhar** [comment]
@ytti yes, will update the PR tomorrow when I have more time
@jgroom33 could be, not sure if this is really the fix; docker (or oxidized, actually) should clean up the PID upon shutdown..

**Comment by Exhar** [comment]
Updated the request; but am not sure if this went as expected ;)

Not sure if you can pick only my changes or not. If not, Ill try to figure out how to solve this... (sigh :))

**Comment by ytti** [comment]
You probably want File.expand_path as ~/ won't be something we can write or read.

**Comment by ElvinEfendi** [comment]
To keep what we have currently as default you just need to do:
```
asetus.default.pid = File.join(Oxidized::Config::Root, 'pid')
```

**Comment by ytti** [comment]
+1

**Comment by ytti** [comment]
@Exhar any update?

**Comment by ytti** [comment]
@Exhar pong?

**Comment by danilopopeye** [comment]
@ytti this can be closed after #435 was merged, right?

**Comment by ytti** [comment]
Yes, thanks.

**Pull request closed without merging**